qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+from qiskit.quantum_info import Statevector, DensityMatrix
+
+from qiskit_dynamics import Solver
+from qiskit_dynamics.array import Array
+from qiskit_dynamics.models import HamiltonianModel
+
+from .dynamics_job import DynamicsJob
+from .pulse_utils import _get_dressed_state_decomposition, _get_lab_frame_static_hamiltonian
+
+
+class PulseSimulator(BackendV2):
+    def __init__(
+        self,
+        solver: Solver,
+        subsystem_dims,
+        subsystem_labels: Optional[List[int]] = None,
+        name: Optional[str] = 'PulseSimulator',
+    ):
+        """This init needs fleshing out. Need to determine all that is necessary for each use case.
+
+        Assumptions
+            - Solver is well-formed.
+            - The no-rotating frame Solver Hamiltonian operators are specified in undressed
+              basis using standard tensor-product convention (whether dense or sparse evaluation)
+
+        Design questions
+            - Simulating measurement requires subsystem dims and labels, should we allow this
+              to be constructed without them, and then measurement is just not possible?
+            - Should we add the ability to do custom measurements?
+            - Should we fill out defaults by extracting them from Solver?
+            - If no set frequency commands are set in the schedule, what should the channel
+              frequencies be?
+                - How are control channel frequencies handled these days?
+            - Fill out properties
+
+        Notes:
+            - Add validation of the Solver object, verifying that its configured to simulate pulse
+              schedules
+        """
+        ##############################################################################################
+        # what to put for provider?
+        super().__init__(
+            provider=None,
+            name=name,
+            description='Pulse enabled simulator backend.',
+            backend_version=0.1
+        )
+
+        self._solver = solver
+        self._subsystem_dims = subsystem_dims or [solver.model.dim]
+        self._subsystem_labels = subsystem_labels or list(range(len(subsystem_dims)))
+
+        # get the dressed states
+        static_hamiltonian = _get_lab_frame_static_hamiltonian(solver.model)
+        dressed_evals, dressed_states = _get_dressed_state_decomposition(static_hamiltonian)
+        self._dressed_evals = dressed_evals
+        self._dressed_states = dressed_states
+        self._dressed_states_adjoint = self._dressed_states.conj().transpose()
+
+    @property
+    def solver(self):
+        """The internal Solver."""
+        return self._solver
+
+    @property
+    def subsystem_dims(self):
+        """Subsystem dimensions."""
+        return self._subsystem_dims
+
+    @property
+    def subsystem_labels(self):
+        """Subsystem labels."""
+        return self._subsystem_labels
+
+    def run(
+        self,
+        run_input: Union[Schedule, ScheduleBlock],
```

Comment:
This should simulate circuits with schedules embedded in the calibrations.
```suggestion
        run_input: Union[QuantumCircuit, List[QuantumCircuit]],
```
IIRC backends should not run schedules directly but instead run circuits with schedules in them.

Reply:
This should be union of schedules and quantum circuits (single or lists)

Reply:
This has been updated to handle circuits, schedules, and scheduleblocks. If a circuit, it will attempt to transpile it to a schedule. If the user has supplied enough information (either by attaching a pulse definition ot individual gates, or having basis gates defined in the target) it will succeed, otherwise the scheduler will raise an informative error itself.